projects/scuba-physics/src/lib/algorithm/BuhlmannAlgorithm.ts

```diff
@@ -300,7 +310,10 @@ export class BuhlmannAlgorithm {
 
     private ascentToNextStop(context: AlgorithmContext, nextStop: number): void {
         const depthDifference = context.currentDepth - nextStop;
-        const duration = durationFor(depthDifference, context.ascentSpeed);
+        let duration = durationFor(depthDifference, context.ascentSpeed);
```

Comment:
I dont think it is enough. What if the ascent from 21m to 12 m with speed 6 m/min.
You will round every 3m segment, instead round only once, e.g. now your ascent will be 3 min (3*Round(0.5)).
Expected is 2 min (Round(3*0.5)).